fendermint/testing/materializer/tests/docker_tests/layer_2_plus.rs

```diff
@@ -0,0 +1,434 @@
+    }
+}
+
+use fendermint_materializer::{manifest::Rootnet, materials::DefaultAccount};
+
+pub type TestMiddleware<C> = SignerMiddleware<Provider<C>, Wallet<SigningKey>>;
+
+async fn make_middleware<C>(
+    provider: Provider<C>,
+    sender: &DefaultAccount,
+) -> anyhow::Result<TestMiddleware<C>>
+where
+    C: JsonRpcClient,
+{
+    let chain_id = provider
+        .get_chainid()
+        .await
+        .context("failed to get chain ID")?;
+
+    let wallet: Wallet<SigningKey> = Wallet::from_bytes(sender.secret_key().serialize().as_ref())?
+        .with_chain_id(chain_id.as_u64());
+
+    Ok(SignerMiddleware::new(provider, wallet))
+}
+
+#[serial_test::serial]
+#[tokio::test]
+async fn test_jedu_rajenka() {
+    // cast logs -r "http://localhost:30945" --address 0x77aa40b105843728088c0132e43fc44348881da8
+    let root_provider = Arc::new(Provider::<Http>::try_from("http://localhost:30845").unwrap());
+    let england_provider = Arc::new(Provider::<Http>::try_from("http://localhost:30945").unwrap());
+    let contract_address = H160::from_str("0x77aa40b105843728088c0132e43fc44348881da8").unwrap();
+
+    let england_gateway_getter = GatewayGetterFacet::new(contract_address, england_provider.clone());
+
+    // let finality: ParentFinality = england_gateway_getter
+    //     .get_latest_parent_finality()
+    //     .call()
+    //     .await
+    //     .unwrap();
+
+    // let block_hash = hex::encode(finality.block_hash);
+    // println!("England finality: {:?} {:?}", finality.height, block_hash);
+
+    let england_gateway_messenger = GatewayMessengerFacet::new(
+        contract_address,
+        england_provider.clone(),
+    );
+
+    let filter = Filter::new().address(contract_address);
+
+    // Query the logs
+    let logs = england_provider.get_logs(&filter).await.unwrap();
+
+    // Process each log by parsing it into the event struct
+    for log in logs {
+        println!("Historical Event: {:?}", log);
+    }
+
+    // let events = england_gateway_messenger
+    //     .events()
+    //     .from_block(1)
+    //     .to_block(ethers::types::BlockNumber::Latest)
+    //     .query()
+    //     .await
+    //     .unwrap();
+
+    // println!("---- Events: {:?}", events.len());
+
+    // events.iter().for_each(|event| {
+    //     println!("---- Event: {:?}", event);
+    // });
+
+    // let root_gateway_messenger = GatewayMessengerFacet::new(
+    //     contract_address,
+    //     root_provider.clone(),
+    // );
+
+    // let events: Vec<NewTopDownMessageFilter> = root_gateway_messenger
+    //     .new_top_down_message_filter()
+    //     .from_block(1)
+    //     .to_block(ethers::types::BlockNumber::Latest)
+    //     .query()
+    //     .await
+    //     .unwrap();
+
+    // println!("---- Events: {:?}", events.len());
+
+    // events.iter().for_each(|event| {
+    //     println!("---- Event: {:?}", event);
+    // });
+}
+
+#[serial_test::serial]
+#[tokio::test]
+async fn test_cross_messages() {
```

Comment:
It would be very helpful to add some simple description for what the test is about, from reading the code, it's quite difficult to infer.